button-merchant/src/main/java/com/usebutton/merchant/ButtonRepositoryImpl.java

```diff
@@ -72,6 +75,11 @@ static ButtonRepository getInstance(ButtonApi buttonApi, PersistenceManager pers
     public void setApplicationId(String applicationId) {
         this.applicationId = applicationId;
         buttonApi.setApplicationId(applicationId);
+
+        for (Task<?> task : pendingTasks) {
+            executorService.submit(task);
```

Comment:
These tasks will fail without an appId. I don't see where you're adding the appId to the requests once we've received it. I just realized I have the [same issue over here](https://github.com/button/button-merchant-ios/pull/61).

Reply:
[Added on iOS](https://github.com/button/button-merchant-ios/pull/61/commits/3fc3df19f6b8b93f2f431604eec7cf2d938d5328)

Reply:
@wessmith The bug you mentioned isn't likely to occur here. The application_id is appended in `ConnectionManager`. That class only comes into play once the `Task` has been submitted -- which only occurs after `configure(~)`.